src/fs/rock/RockRebuild.cc

```diff
@@ -694,19 +697,41 @@ Rock::Rebuild::swanSong()
     storeRebuildComplete(&counts);
 }
 
+void
+Rock::Rebuild::callException(const std::exception &e)
+{
+    ErrorDetail::Pointer errorDetail;
+    if (const auto tex = dynamic_cast<const TextException*>(&e))
+        errorDetail = new ExceptionErrorDetail(tex->id());
+    else
+        errorDetail = new ExceptionErrorDetail(Here().id());
+    failure(ToSBuf(e.what(),
+        Debug::Extra, errorDetail).c_str());
+    // fatal failure() substitutes default AsyncJob::callException(e) handling
```

Comment:
I would start addressing the above TODO with these steps:

* make callExcpetion() create a new exception with e.what() primary text and the following Debug::Extra lines: "context: Rebuilding tock cache_dir index for ..." and "scan progress: ...". These details should be moved from failure(). This assumes that an exception thrown from callException() terminates the process and reports the exception to cache.log (at level 0 or level 1).
* make failure() throw a TextException with msg as the primary text. You already have code for supplying Extra errno-based system call error detail to that exception. Keep that. Do not call fatal(). This assumes that an exception thrown from failure() gets redirected to callException().

Reply:
It looks like we follow another suggested approach, with just 'throw;'.

Reply:
Yes, sorry, my change request was very stale -- I forgot to delete it after refactoring the review.